NuclearData/ceNeutronData/ceNeutronDatabase_inter.f90

```diff
@@ -196,7 +197,31 @@ subroutine updateMicroXSs(self, E, nucIdx, rand)
     end subroutine updateMicroXSs
 
     !!
-    !! Subroutine to get the elastic scattering majorant cross section in a nuclide
+    !! Subroutine to retrieve the nuclide total majorant cross section
+    !! over a calculated energy range (needed for TMS) and update the nuclide
+    !! cache with majorant value, energy, deltakT and Doppler correction factor
+    !!
+    !! The energy range depends on the nuclide and TMS material kT, on the
+    !! atomic mass of the nuclide, and the incident neutron energy
+    !!
+    !! Args:
+    !!   E [in]      -> required energy [MeV]
+    !!   kT [in]     -> Thermal energy of TMS material
+    !!   nucIdx [in] -> material index that needs to be updated
+    !!
+    !! Errors:
+    !!   FatalError if material kT is smaller than the nuclide kT
+    !!
+    subroutine updateTotalTempNucXS(self, E, kT, nucIdx)
```

Comment:
Can this be given a more descriptive name? I suppose for me Temp has an annoying double connotation. Reading the comments, it isn't clear how the name relates to the function other than TMS - maybe updateTotalTMSNucXS?